moto/awslambda/models.py

```diff
@@ -1443,28 +1483,66 @@ def list_aliases(self, function_name: str) -> Iterable[LambdaAlias]:
         return sorted(aliases.values(), key=lambda alias: alias.name)
 
     def get_arn(self, arn: str) -> Optional[LambdaFunction]:
+        [arn_without_qualifier, _, _] = self.split_function_arn(arn)
+        return self._arns.get(arn_without_qualifier, None)
+
+    def split_function_arn(self, arn: str) -> [str, str, Optional[str]]:
```

Comment:
Handy utility to parse an ARN into:
- ARN without qualifier
- function name
- qualifier